faker/providers/ssn/en_US/__init__.py

```diff
@@ -1,19 +1,172 @@
-        area = self.random_int(min=1, max=899)
-        if area == 666:
-            area += 1
-        group = self.random_int(1, 99)
-        serial = self.random_int(1, 9999)
+        # The group number must be between 70 and 99 inclusively but not 89 or 93
+        group = random.choice([x for x in range(70,100) if x not in [89, 93]])
 
         ssn = "{0:03d}-{1:02d}-{2:04d}".format(area, group, serial)
-        return ssn
+        return(ssn)
+
+    def ein(self):
+        """Generate a random United States Employer Identification Number (EIN).
+        
+         An United States An Employer Identification Number (EIN) is
+         also known as a Federal Tax Identification Number, and is
+         used to identify a business entity. EINs follow a format of a
+         two-digit prefix followed by a hyphen and a seven-digit sequence:
+         ##-######
+
+         https://www.irs.gov/businesses/small-businesses-self-employed/employer-id-numbers
+        """
+
+        # Only certain EIN Prefix values are assigned:
+        #
+        # https://www.irs.gov/businesses/small-businesses-self-employed/how-eins-are-assigned-and-valid-ein-prefixes
+
+        ein_prefix_choices = [
+            '01',
+            '02',
+            '03',
+            '04',
+            '05',
+            '06',
+            '10',
+            '11',
+            '12',
+            '13',
+            '14',
+            '15',
+            '16',
+            '20',
+            '21',
+            '22',
+            '23',
+            '24',
+            '25',
+            '26',
+            '27',
+            '30',
+            '31',
+            '32',
+            '33',
+            '34',
+            '35',
+            '36',
+            '37',
+            '38',
+            '39',
+            '40',
+            '41',
+            '42',
+            '43',
+            '44',
+            '45',
+            '46',
+            '47',
+            '48',
+            '50',
+            '51',
+            '52',
+            '53',
+            '54',
+            '55',
+            '56',
+            '57',
+            '58',
+            '59',
+            '60',
+            '61',
+            '62',
+            '63',
+            '64',
+            '65',
+            '66',
+            '67',
+            '68',
+            '71',
+            '72',
+            '73',
+            '74',
+            '75',
+            '76',
+            '77',
+            '80',
+            '81',
+            '82',
+            '83',
+            '84',
+            '85',
+            '86',
+            '87',
+            '88',
+            '90',
+            '91',
+            '92',
+            '93',
+            '94',
+            '95',
+            '98',
+            '99']
+
+        ein_prefix = random.choice(ein_prefix_choices)
+        sequence = self.random_int(min=0, max=9999999)
+
+        ssn = "{0:s}-{1:07d}".format(ein_prefix, sequence)
```

Comment:
this is not an SSN, but an EIN. So this variable should be called `ein`

Reply:
Good catch.